### ISSUE TITLE
clean up the source after stopping

### ISSUE THREAD

**Issue by dominictarr** [Fishrock123/bob#58]
this module implements a stop method:
https://github.com/Fishrock123/bob/blob/master/reference-extension-stop.js

but it doesn't notify the source in anyway. If the source is using a heavy resource, (eg, a file descriptor) the source needs to know the sink has stopped so that it can close the resource.

**Comment by Fishrock123** [comment]
Right, that file is a reference. The source would have to implement it itself.
